resolver/src/resolver.rs

```diff
@@ -84,8 +84,8 @@ impl Resolver {
 
     /// Constructs a new Resolver with the system configuration.
     ///
-    /// This will read the systems `/etc/cresolv.conf`. Only Unix like OSes are currently supported.
-    #[cfg(unix)]
+    /// This will read the systems `/etc/cresolv.conf` on Unix OSes.
+    #[cfg(not(all(target_os = "windows", target_pointer_width = "32")))]
```

Comment:
Is there a way to leave this as `unix`? It might end up being clearer logically to have a single method, with and if-else-if chain using: https://doc.rust-lang.org/std/macro.cfg.html

Then we could default to failed in the last else statement, and what is used becomes more clear...

Reply:
Are you sure?
That means the error will be at runtime instead of compile time (the way it's now the function simply doesn't exist for windows 32-bit).

Reply:
You’re right. I wasn’t thing. Is it possible to make this case positive instead of negative though? Or is there a case where unix gets triggered on Windows?

Reply:
It could be `#[cfg(any(unix, all(target_os = "windows", target_pointer_width = "64")))]`, do you prefer that?
And what exactly do you mean by "a case where unix gets triggered on Windows"?

Reply:
I think we can leave it for now. Any OS' where it's not supported will crop up in the future and we can fix then...